ui/components/grant.js

```diff
@@ -63,23 +64,15 @@ export default function ({ id }) {
   const handleShareGrant = async () => {
     setError('')
     try {
-      let res = await fetch(`/api/users?name=${name}`)
+      const res = await fetch(`/api/users?name=${name}`)
       const data = await res.json()
 
       if (!res.ok) {
         throw data
       }
 
       if (data?.items?.length === 0) {
-        res = await fetch('/api/users', {
-          method: 'POST',
-          body: JSON.stringify({ name })
-        })
-        const user = await res.json()
-
-        await grantPrivilege(user.id)
-        setName('')
-        setRole('view')
+        setError('User does not exist')
```

Comment:
We should warn that the user does not exist, but allow the grant to be created anyway. The user not being known to Infra will be a common case as long as we are doing identities "just in time" from identity providers.

Reply:
but i think for the current workflow, we will not be able to get the one time password if the user create a user on the grant view. I think what we should do is maybe come up with a make sense workflow for that and before that we should prevent user to create a user on the grant

Reply:
That's okay: for now we'll restrict this, and handle JIT user capabilities in a future iteration (e.g. a prompt, checkbox or similar - similar to the CLI's `-f` flag). For now users will have to log in to get granted access